bootstrap/components/sinks/kafka-sink-topology-component.json

```diff
@@ -35,13 +35,23 @@
         "hint": "schema,override"
       },
       {
-        "uiName": "Writer schema version",
+        "uiName": "Writer Schema Branch",
+        "fieldName": "schemaBranch",
+        "isOptional": true,
+        "tooltip": "Schema branch to be used fetching for writer schema versions. Default branch is MASTER",
+        "type": "enumstring",
+        "options": [],
+        "hint": "schema,override",
+        "defaultValue": "MASTER"
+      },
+      {
+        "uiName": "Writer Schema Version",
         "fieldName": "writerSchemaVersion",
         "isOptional": false,
         "tooltip": "Version of schema for topic to use for serializing the message. Default is the latest version for the schema",
         "type": "enumstring",
         "options": [],
-        "hint": "dependsOn-topic,schemaVersion"
+        "hint": "dependsOn-topic,dependsOn-schemaBranch,schemaVersion"
```

Comment:
Probably remove dependsOn-topic here if its a selection in order. I would imagine that user will first pick topic that UI will use to get all branches and then once user selects the branch we get all versions for the branch.  In that case schemaVersion can now only depend on schemaBranch. However, I see that schemaBranch is optional field. In that case it makes sense.